polymorphia-frontend/components/xp-card/components/RoadmapCardRightComponent.tsx

```diff
@@ -0,0 +1,39 @@
+import { useUserDetails } from "@/hooks/contexts/useUserContext";
+import { TargetTypes } from "@/interfaces/api/target";
+import useShortGrade from "@/hooks/course/useShortGrade";
+import XPCardPoints from "@/components/xp-card/components/XPCardPoints";
+import { RoadmapCardRightComponentProps } from "@/app/(logged-in)/roadmap/types";
+
+export default function RoadmapCardRightComponent({
+  gradableEvent,
+}: RoadmapCardRightComponentProps) {
+  const { id: userId } = useUserDetails();
+  const target = {
+    id: userId,
+    type: TargetTypes.STUDENT,
+  };
+  const { data: grade, isLoading: isLoading } = useShortGrade(
+    target,
+    gradableEvent.id
+  );
+
+  if (isLoading || !grade) {
```

Comment:
```suggestion
  const { data: grade, isLoading: isLoading, isError: isError } = useShortGrade(
    target,
    gradableEvent.id
  );

  if (isLoading || !grade || isError) {
```
I don't have an idea for better handling the error here. But if there is backend error, it should be displayed in a toast anyway, maybe that'd be enough